src/regmod/prior.py

```diff
@@ -62,11 +208,59 @@ def __post_init__(self):
             assert self.size == self.mat.shape[0], "`mat` and `size` not match."
 
     def is_empty(self) -> bool:
+        """Indicate if the prior is empty.
+
+        Returns
+        -------
+        bool
+            Return `True` if `self.mat` is empty.
+        """
         return self.mat.size == 0.0
 
 
 @dataclass
 class SplinePrior(LinearPrior):
+    """Spline prior information.
+
+    Parameters
+    ----------
+    size : int, default=100
+        Size of the spline prior. Default is 100. It determines the number
+        of sample points in the specified domain.
+    order : int, default=0
+        Order of the spline derivative. Default is 0.
+    domain_lb : float, default=0.0
+        Lower bounds of the domain. Default is 0.0.
+    domain_ub : float, default=1.0
+        Upper bounds of the domain. Default is 1.0.
+    domain_type : {'abs', 'rel'}, default='rel'
```

Comment:
When the options are listed, I think the default typically comes first (see https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard, section 4)